From 5985bb59b193a7e014b3a26b616f81310c91816b Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 15 Feb 2022 12:24:12 +0200
Subject: [PATCH] [SW-75322] greco: fw_loader: fix unknown reset type

After firmware update the hl_fw_loader issues firmware
all reset and this reset type is missing for greco and
ppboot was printing reset cause as unknown hence adding
all the supported reset types for greco.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Id9f6beb80f86dc77d9be4a176e7fe2a20a1fddc3
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 706bca9..051b41d 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -105,6 +105,44 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
 }
 
+void rst_src_dump(void)
+{
+	unsigned int rst_src = hl_rst_src_get();
+	char *cause;
+
+	switch (rst_src) {
+	case HL_COLD_RST:
+		cause = "cold";
+		break;
+	case HL_MANUAL_RST:
+		cause = "manual";
+		break;
+	case HL_PRSTN_RST:
+		cause = "prstn";
+		break;
+	case HL_SOFT_RST:
+		cause = "soft";
+		break;
+	case HL_WD_RST:
+		cause = "watch dog";
+		break;
+	case HL_FW_ALL_RST:
+		cause = "firmware all";
+		break;
+	case HL_SW_ALL_RST:
+		cause = "software all";
+		break;
+	case HL_FLR_RST:
+		cause = "functional";
+		break;
+	default:
+		cause = "unknown";
+		break;
+	}
+	preboot_puts_raw("Last reset cause is: ");
+	preboot_puts_raw(cause);
+	preboot_puts(" reset\n");
+}
 void preboot_init_pcie_phy_fw_update(void)
 {
 	/* Configure FW update size */
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index 84d1b90..53982e6 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -43,9 +43,12 @@
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
-	HL_PRSTN_RST = 2,
-	HL_WD_RST = 4,
+	HL_PRSTN_RST,
+	HL_SOFT_RST,
+	HL_WD_RST,
+	HL_FW_ALL_RST = 5,
 	HL_SW_ALL_RST = 8,
+	HL_FLR_RST = 9,
 };
 
 void run_arc(u8 id);
